### PR TITLE
py-spopt: update to 0.6.0

### PR DESCRIPTION
**Pull request by nilason**
###### Type(s)

- [ ] bugfix
- [ ] enhancement
- [ ] security fix

###### Tested on

macOS 14.2.1 23C71 arm64
Xcode 15.1 15C65

###### Verification 
Have you

- [x] followed our [Commit Message Guidelines](https://trac.macports.org/wiki/CommitMessages)?
- [x] squashed and [minimized your commits](https://guide.macports.org/#project.github)?
- [x] checked that there aren't other open [pull requests](https://github.com/macports/macports-ports/pulls) for the same change?
- [ ] referenced existing tickets on [Trac](https://trac.macports.org/wiki/Tickets) with full URL? 
- [ ] checked your Portfile with `port lint --nitpick`?
- [x] tried existing tests with `sudo port test`?
- [x] tried a full install with `sudo port -vst install`?
- [ ] tested basic functionality of all binary files?
- [ ] checked that the Portfile's most important [variants](https://trac.macports.org/wiki/Variants) haven't been broken?

### CODE REVIEW

**Comment by nilason** [comment]
I dropped the commit to add py312 support to all PySAL port because of lacking dependencies:

- py312-geopandas
- py312-scipy
- py312-scikit-learn
- py312-quantecon
- py312-joblib
- py312-seaborn
- py312-numba
- py312-rtree
- py312-rasterio
- py312-statsmodels
- py312-rasterstats
- py312-pulp
- py312-codecov
- py312-folium

File: python/py-spopt/Portfile [hunk 1]
```diff
@@ -26,15 +25,15 @@ long_description    {*}${description} is an open-source Python library for \
...
 
-checksums           rmd160  4d387eb900763e5e49b41b1f68ace0c0aa3a029e \
-                    sha256  c3d6572c1167b5d59a98781f8697cc6ba10f1fbf65ecf1ca1bad1aaa9d4ce511 \
-                    size    29182492
+checksums           rmd160  897ce8838c0b0224b03c2535f28c455ce2777dfc \
+                    sha256  1edf615a8ab1bbae3881c742083e20f18d24724947e84bf2dd12ec38510bcf3a \
+                    size    30545521
 
 python.versions     39 310 311
+python.pep517       yes
```
Comment (reneeotten) [inline python/py-spopt/Portfile:33]: this is the default for Python 3.7+ and could/should thus be removed

File: python/py-spopt/Portfile [hunk 2]
```diff
@@ -60,6 +59,17 @@ if {${name} ne ${subport}} {
...
+        checksums       rmd160  4d387eb900763e5e49b41b1f68ace0c0aa3a029e \
+                        sha256  c3d6572c1167b5d59a98781f8697cc6ba10f1fbf65ecf1ca1bad1aaa9d4ce511 \
+                        size    29182492
+
+        depends_build-append port:py${python.version}-setuptools
+    }
+
     test.run        yes
     test.cmd        py.test-${python.branch}
     test.target
```
Comment (reneeotten) [inline python/py-spopt/Portfile:75]: the `test.cmd` and `test.target` are set to the same values as is done by default through the `python` PG - where `test.run` uses as default `python.test_framework pytest`.

**Review by reneeotten — CHANGES_REQUESTED** [review]

**Comment by nilason** [comment]
@reneeotten I'm not sure how to move this forward for py39. The problem is, with the update to 0.6.0 it now works fine to get source from pypi, which now includes tests, but support is dropped for Py39. Hence, it needs to be pinned to the GH source of version 0.5.0...

**Comment by reneeotten** [comment]
> @reneeotten I'm not sure how to move this forward for py39. The problem is, with the update to 0.6.0 it now works fine to get source from pypi, which now includes tests, but support is dropped for Py39. Hence, it needs to be pinned to the GH source of version 0.5.0...

I agree that is has to be pinned to the earlier version; bur why not the one from PyPI?

**Comment by nilason** [comment]
> I agree that is has to be pinned to the earlier version; bur why not the one from PyPI?

The current distfile is from GH, size and checksums won’t match.

**Comment by nilason** [comment]
https://distfiles.macports.org/py-spopt/

**Comment by reneeotten** [comment]
> > I agree that is has to be pinned to the earlier version; bur why not the one from PyPI?
> 
> The current distfile is from GH, size and checksums won’t match.

ah okay, well you can either leave it as-is then or treat it as a [StealthUpdate](https://trac.macports.org/wiki/PortfileRecipes#stealth-updates)

**Comment by nilason** [comment]
Finally made this work. Ready for review/merge.

File: python/py-spopt/Portfile [hunk 3]
```diff
@@ -60,7 +58,20 @@ if {${name} ne ${subport}} {
                     port:py${python.version}-spaghetti \
                     port:py${python.version}-tqdm
 
+    if {${python.version} == 39} {
+        PortGroup       github 1.0
+
+        github.setup    pysal spopt 0.5.0 v
+
+        python.pep517   no
```
Comment (reneeotten) [inline python/py-spopt/Portfile:66]: why is this set to `no`; subports for Python 3.7+ should ideally build with it set to `yes`? What is the error you see when you leave this to the default?

Comment (pmetzger) [reply python/py-spopt/Portfile:66]: @nilason ?

File: python/py-spopt/Portfile [hunk 4]
```diff
@@ -60,7 +58,19 @@ if {${name} ne ${subport}} {
...
+
+        github.setup    pysal spopt 0.5.0 v
+
+        version         0.5.0
+        revision        0
+        checksums       rmd160  4d387eb900763e5e49b41b1f68ace0c0aa3a029e \
+                        sha256  c3d6572c1167b5d59a98781f8697cc6ba10f1fbf65ecf1ca1bad1aaa9d4ce511 \
+                        size    29182492
+
+        depends_build-append port:py${python.version}-setuptools
```
Comment (reneeotten) [inline python/py-spopt/Portfile:72]: this is no longer explicitly needed with the `setuptools` PEP517 backend

**Comment by nilason** [comment]
I believe I have fixed all issues now.

**Comment by nilason** [comment]
Thanks!

**Pull request merged**